main.go

```diff
@@ -533,6 +594,101 @@ func main() {
 		log.Fatalf("Failed to add Get Wifi Status characteristic to device service: %s", err)
 	}
 
+	// set wifi connection
+	setWifiChar, err := deviceService.NewChar(setWifiUUIDFragment)
+	if err != nil {
+		log.Fatalf("Failed to create set wifi characteristic: %s", err)
+	}
+
+	setWifiChar.Properties.Flags = []string{
+		gatt.FlagCharacteristicEncryptAuthenticatedWrite,
+		gatt.FlagCharacteristicEncryptAuthenticatedRead,
+	}
+
+	setWifiChar.OnWrite(func(c *service.Char, value []byte) (resp []byte, err error) {
+		defer func() {
+			if err != nil {
+				log.Printf("Error setting wifi connection: %s.", err)
+			}
+		}()
+
+		if SSID == "" {
+			log.Println("Please enter SSID")
+		} else {
+			log.Println("Please enter Wifi Password")
+		}
+
+		if SSID == "" {
+			SSID = string(value)
+		} else {
+			WifiPasswordKey = string(value)
+		}
+
+		log.Println(SSID, "SSID", WifiPasswordKey, "key")
+		if SSID != "" && WifiPasswordKey != "" {
```

Comment:
How about we go with Yev's simpler option: just replace the entire list with our new network? No fiddling with priority and so on.

If you're okay with this I'll modify the ticket.

Reply:
Yeah that works